sentry-core/src/test/java/io/sentry/core/SendCachedEventTest.kt

```diff
@@ -0,0 +1,99 @@
+package io.sentry.core
+
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.whenever
+import com.nhaarman.mockitokotlin2.doThrow
+import java.io.File
+import java.nio.file.Files
+import kotlin.test.Test
+import kotlin.test.assertFalse
+
+class SendCachedEventTest {
+  private class Fixture {
+    var hub: IHub? = mock()
+      var logger: ILogger? = mock()
+      var serializer: ISerializer? = mock()
+    var options = SentryOptions()
```

Comment:
why are all of them nullable? `type?` if they get a fresh instance with `mock()`

Reply:
Often I set each one null and write a test to see how the mock behaves during construction time

Reply:
Didn't do it now because I'm rushing everything*